### PR TITLE
Jf create comment

### LINKED ISSUE #28: Create a Comment
As a commenter, I would like to be able to add a Comment to a Post so that I can take part in the discussion about a Post.

**Given** the user is viewing the Details of a Post
**When** they select the `Add Comment` menu option
**Then** they should be directed to a page with a form for creating a new Comment

**Given** the user has entered the relevant information for a Comment
**When** they click the `Save` button
**Then** the Comment should be saved to the database
**And** the creation datetime should be automatically set to the current date and time
**And** the user should be redirected to the new Comments list page for the related Post  

The "relevant information" for a Comment is

* Subject
* Content

### PR DESCRIPTION
**Pull request by JacobF94**
#### Changes Made
1. Added Comment.cs, CommentRepository.cs, CommentController.cs, ICommentController.cs to server side
2. Added CommentForm.js and CommentManager.js to client side
3. modified PostDetails.js and ApplicationViews.js
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout jf-commentCreate
    ```
4. Open a new Terminal tab (⌘T) and navigate to the server directory.
5. Test app functionality.
    > When logged in as an authenticated user, navigate to the detail page for a post
    > There should be an 'Add Comment' button at the bottom of the page
    > Click on the button, and you should be redirected to a Comment Create form
    > The submit button should be disabled until both fields have text in them
    > After entering in text into the fields and clicking submit, the user should be taken back to the homepage
    > In order to test, you must check your actual database in VS to see if the new comment was created, rendering comments on the detail page will be my next issue

6. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

closes #28

### CODE REVIEW

File: Tabloid/client/src/components/Comments/CommentForm.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    commentCopy[key] = value;
+    setComment(commentCopy);
+  };
+  const handleSave = (evt) => {
+    evt.preventDefault();
+    const commentCopy = { ...comment };
+    commentCopy.PostId = parseInt(postId);
+    addComment(commentCopy).then(() => {
+      history.push("/");
```
Comment (DougSilvera) [inline Tabloid/client/src/components/Comments/CommentForm.js]: history.push should send the user back to the post detail page for the commented post

Comment (JacobF94) [reply Tabloid/client/src/components/Comments/CommentForm.js]: I have addressed this issue in my new push

**Review by DougSilvera — APPROVED** [review]

File: Tabloid/client/src/modules/commentManager.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { getToken } from "./authManager";
+
+const baseUrl = '/api/Comment';
+
+// export const getAllPostComments = (id) => {
```
Comment (heymonicakay) [inline Tabloid/client/src/modules/commentManager.js]: Should we remove this unused code? looks like it will be useful in a future pr, but it doesn't look like it's relevant to this specific pr

File: Tabloid/client/src/components/Comments/CommentForm.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { useHistory } from "react-router-dom";
+import { FormGroup, Input, Label, Button, Form } from "reactstrap";
+import { addComment } from "../../modules/commentManager";
+import { useParams } from "react-router-dom";
+
+const CommentForm = () => {
+  const history = useHistory();
+  const { postId } = useParams();
+  const emptyComment = {
+    PostId: 1,
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Comments/CommentForm.js]: Since there is a good possibility that there is actually a post with the id of 1, should this initial value actually be 0 instead? 0 is sort of the numeric equivalent of "nothing", which is usually what the initial state is set to...

File: Tabloid/client/src/components/Comments/CommentForm.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { FormGroup, Input, Label, Button, Form } from "reactstrap";
+import { addComment } from "../../modules/commentManager";
+import { useParams } from "react-router-dom";
+
+const CommentForm = () => {
+  const history = useHistory();
+  const { postId } = useParams();
+  const emptyComment = {
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Comments/CommentForm.js:10]: watch your casing.. your client side variables should be camel cased unless they are components

File: Tabloid/client/src/components/Comments/CommentForm.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    addComment(commentCopy).then(() => {
+      history.push(`/posts/${postId}`);
+    });
+  };
+
+  const SubmitButton = () => {
+    if (comment.Subject && comment.Content){
+      return <Button className="btn btn-primary" onClick={handleSave}>Submit</Button>
+    } else {
+      return <button type="button" disabled>Button</button>
```
Comment (heymonicakay) [inline Tabloid/client/src/components/Comments/CommentForm.js]: Is there a reason for using a regular button element here instead of the Button component from react strap? 

https://react-bootstrap.github.io/components/buttons/#disabled-state

File: Tabloid/Repositories/CommentRepository.cs [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                    DbUtils.AddParameter(cmd, "@Subject", comment.Subject);
+                    DbUtils.AddParameter(cmd, "@Content", comment.Content);
+                    DbUtils.AddParameter(cmd, "@CreateDateTime", comment.CreateDateTime);
+
+                    comment.Id = (int)cmd.ExecuteScalar();
+                }
+            }
+        }
+
+        public void Delete(int id)
```
Comment (heymonicakay) [inline Tabloid/Repositories/CommentRepository.cs]: Looks like this is unrelated to this issue ticket. Please remove this code until the appropriate issue is being addressed.

File: Tabloid/Repositories/CommentRepository.cs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+                conn.Open();
+                using (var cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = "DELETE FROM Comment WHERE Id = @Id";
+                    DbUtils.AddParameter(cmd, "@id", id);
+                    cmd.ExecuteNonQuery();
+                }
+            }
+        }
+        public void Update(Comment comment)
```
Comment (heymonicakay) [inline Tabloid/Repositories/CommentRepository.cs]: Looks like this is unrelated to this issue ticket. Please remove this code until the appropriate issue is being addressed.

File: Tabloid/Controllers/CommentController.cs [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        public IActionResult Post(Comment comment)
+        {
+            comment.CreateDateTime = DateTime.Now;
+            UserProfile currentUser = GetCurrentUserProfile();
+            comment.UserProfileId = currentUser.Id;
+            _commentRepo.Add(comment);
+            return Ok(comment);
+        }
+
+        [HttpPut("{id}")]
```
Comment (heymonicakay) [inline Tabloid/Controllers/CommentController.cs]: Looks like this is unrelated to this issue ticket. Please remove this code until the appropriate issue is being addressed.

File: Tabloid/Controllers/CommentController.cs [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+            if (id != comment.Id)
+            {
+                return BadRequest();
+            }
+
+            _commentRepo.Update(comment);
+            return NoContent();
+        }
+
+        [HttpDelete("{id}")]
```
Comment (heymonicakay) [inline Tabloid/Controllers/CommentController.cs]: Looks like this is unrelated to this issue ticket. Please remove this code until the appropriate issue is being addressed.

**Review by heymonicakay — CHANGES_REQUESTED** [review]
Please see my comments

File: Tabloid/client/src/components/Comments/CommentForm.js (hunk 4 above)
Comment (JacobF94) [reply Tabloid/client/src/components/Comments/CommentForm.js:10]: Isn't CommentForm.js a component?

**Review by heymonicakay — APPROVED** [review]
Thanks @JacobF94 !
